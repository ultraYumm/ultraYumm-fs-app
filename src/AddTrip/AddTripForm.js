import React, { Component } from 'react';
import './AddTrip.css';
import '../Font/Font.css';
import GoButton from '../FormElements/GoButton';
import '../FormElements/FormElements.css';



class AddTripForm extends Component {

  
  render() {
    
    const onSubmitForm = () => {
      
    }
  

       
   
    return (
        <form onSubmit={onSubmitForm}>
         <h2>Plan a trip!</h2>
            <p>
                <label htmlFor= "new-trip-name"><span className= "labelWidth">New trip Name</span>
                    <input type="text" name="newTripName" className="salmonBackground white search" id= "new-trip-name" placeholder="Escalante" required/>
                </label>
            </p>

            <p>
                <label htmlFor= "start-date"><span className= "labelWidth">Start date</span>
                <input type="date" name="startDate" id= "start-date"/> </label> <br></br>
                <label htmlFor= "end-date"><span className= "labelWidth">End date</span>
                <input type="date" name="endDate" id= "end-date"/> </label>            
            </p>

            <p>
                 <label htmlFor= "traveler-name"><span className= "labelWidth">  Traveler Names</span>
                 <input type="text" name="traveler-name" className= "mauveBackground white names" placeholder = "Stef, Jack, Emi, Marielle..." id= "traveler-name"/>  
                </label>
            </p>

            <p>
                 <label htmlFor= "map-link"><span className= "labelWidth"><a href= "google.com/maps" target= "_blank">Link</a></span>
                 <input type="url" name="map-link" className= "mauveBackground white names" placeholder = "https://www.google.com/maps/embed?pb=!4v1589308620339!6m8!1m7!1sH9R4rDjmbhhsQVA2f0Dq8Q!2m2!1d36.03291467946964!2d-111.8531019810478!3f9.17!4f0!5f0.7820865974627469" id= "map-link"/>  
                </label>
            </p>


          <GoButton></GoButton>
        </form>
        
   )}
  
}
  

export default AddTripForm