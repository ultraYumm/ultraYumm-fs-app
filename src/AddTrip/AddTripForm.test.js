import React from 'react';
import ReactDOM from 'react-dom';
import AddTripForm from './AddTripForm';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <BrowserRouter>
     <AddTripForm />
  </BrowserRouter> 
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create( <BrowserRouter>
         <AddTripForm />
         </BrowserRouter> )
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  