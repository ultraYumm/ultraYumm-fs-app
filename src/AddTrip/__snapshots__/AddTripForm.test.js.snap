// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <h2
    className="white"
  >
    <i
      className="fas fa-shoe-prints"
    />
     Plan a trip! 
    <div
      className="goContainer go"
    >
      <button
        className="white montebello blackBackground go"
        type="submit"
      >
        Go!
      </button>
    </div>
  </h2>
  <div
    className="labelWidthPlan"
  >
    <label
      htmlFor="new-trip-name"
    >
      <i
        className="fas fa-feather white"
      />
      <span
        className="labelWidthPlan white montebello"
      >
        New trip name
      </span>
      <input
        className="skinBackground black search"
        defaultValue="my awesome trip"
        id="new-trip-name"
        name="tripName"
        onChange={[Function]}
        type="text"
      />
      <span
        className="error cloudBlue"
      >
        
      </span>
    </label>
  </div>
  <div
    className="labelWidthPlan"
  >
    <label
      htmlFor="traveler-name"
    >
      <i
        className="fas fa-user-friends white"
      />
      <span
        className="labelWidthPlan white montebello"
      >
        Traveler names
      </span>
      <input
        className="skinBackground purple names"
        defaultValue={
          Array [
            "Lewis",
            " Clark",
            " Alexandra",
            "TBD",
          ]
        }
        id="traveler-name"
        name="tripTravelers"
        onChange={[Function]}
        type="text"
      />
      <span
        className="error cloudBlue"
      >
        
      </span>
    </label>
  </div>
  <div
    className="labelWidthPlan"
  >
    <label
      htmlFor="map-link"
    >
      <i
        className="fas fa-drafting-compass white"
      />
      <span
        className="labelWidthPlan white"
      >
        <a
          className="white montebello"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Website
        </a>
      </span>
      <input
        className="skinBackground purple names"
        defaultValue="https://earth.google.com"
        id="map-link"
        name="tripURL"
        type="url"
      />
    </label>
  </div>
  <br />
  <div
    className="dates"
  >
    <div
      className="labelWidthDates"
    >
      <label
        htmlFor="start-date"
      >
        <span
          className="white montebello labelWidthPlan"
        >
          Start date
        </span>
        <input
          id="start-date"
          name="startDate"
          onChange={[Function]}
          type="date"
        />
         
      </label>
       
      <br />
    </div>
    <div
      className="labelWidthDates"
    >
      <label
        htmlFor="end-date"
      >
        <span
          className="white montebello labelWidthPlan"
        >
          End date
        </span>
        <input
          id="end-date"
          name="endDate"
          onChange={[Function]}
          type="date"
        />
        <span
          className="error cloudBlue"
        />
      </label>
    </div>
  </div>
</form>
`;
